libraries/standard/http/include/http_client.h

```diff
@@ -0,0 +1,339 @@
+#ifndef HTTP_CLIENT_H_
+#define HTTP_CLIENT_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @brief Maximum amount of headers allowed from the server.
+ * 
+ * If the server sends more headers than this configuration, then status code
+ * #HTTP_SECURITY_ALERT_TOO_MANY_HEADERS is returned from #HTTPClient_Send.
+ */
+#ifndef HTTP_MAX_HEADERS_BYTES
+    #define HTTP_MAX_HEADERS_BYTES      2048U
+#endif
+
+/**
+ * @brief The HTTP header "User-Agent" value.
+ * 
+ * The following headerline is automatically written to
+ * #HTTPRequestHeaders_t.pBuffer:
+ * "User-Agent: AWS\r\n"
+ */ 
+#ifndef HTTP_USER_AGENT_VALUE
```

Comment:
Is there a reason for not breaking this into multiple header files to separate the functions and types? Also, there are a number of constant macros which could be moved to the "types" specific header file. 
(Though a subjective preference,) I think it would be better for file readability

Reply:
MQTT is not doing that right now. We can decide to this later.

Reply:
Breaking up to multiple header files can be done later if we decide to this.